packages/core/src/Parser.ts

```diff
@@ -232,7 +239,11 @@ export class Parser {
      * @param container
      * @param offset
      */
-    _locate(container: Node, offset: number): [VNode, RelativePosition] {
+    _locate(
+        vDocumentMap: VDocumentMap,
+        container: Node,
+        offset: number,
+    ): [VNode, RelativePosition] {
```

Comment:
Same for the Parser, I would gave the parser its own map rather than pollute the signature of all functions to pass the map to them.